WordPress/gutenberg
# Optimize preload paths for post and site editors

This PR optimizes the preloaded data for REST endpoints. Some preloaded data are removed, some endpoints are modified, some are added.

**Removing the `/` preload.** One of the endpoints we preload is `/`, where we use properties like `site_icon_url` or `home`. The response, however, also contains a [huge `routes` field](https://github.com/WordPress/gutenberg/issues/27370), bloating the initial HTML response. Before we can filter them out with a `_fields` query param, some [Core patches](https://core.trac.wordpress.org/ticket/55213) will need to be merged first and that will take time.

In this PR I'm simply removing the preload. It's not needed for anything critical and the editor is perfectly functional even without the data. The data are used, for example, to display the site logo in place of the WP logos here (the fullscreen-close and pre-publish panels):

<img width="801" alt="Screenshot 2022-03-07 at 15 28 33" src="https://user-images.githubusercontent.com/664258/157054572-f8ff74e8-61d2-4340-ba7b-2db5e2903061.png">

That can certainly wait and doesn't need to block the editor initialization.

**Fixing preloads for `postType` and `taxonomies` entities.** There are functions that load entity data of certain types ([`postType`](https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/entities.js#L208) and [`taxonomies`](https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/entities.js#L243)), required to resolve any selector for these entities. They call the endpoints with `context=view`, but the preloads are for `context=edit` and are not used. This PR fixes the `context` params to match the real requests. An alternative way to fix this would be to change the actual requests to `context=edit`. I'm not sure what's the difference and whether it matters. Up for discussion.

**Fixing preload for `users/me`.** The preload is using a [wrong URL](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/edit-form-blocks.php#L66):
```
/wp/v2/users/me?post_type=%s&context=edit
```
but we're fetching from just
```
/wp/v2/users/me
```
The old URL hasn't been used for a very long time: it used to load user permissions for a particular post type, and was used to implement the `canUser` selector. But we've been using `OPTIONS` requests and `Allow` response headers for that for a long time.

**Adding preload for `/settings`.** The request is done on a critical path, in the `useBlockEditorSettings` hook, so all the settings had better be available as soon as possible.

**How to test:**
Verify in the Network Devtool that none of these preloaded paths trigger a network request -- that would mean that the preloaded cached version was missed.

Verify that the initial HTML got significantly smaller as there are no longer data for the `/` endpoint.
